### ISSUE TITLE
Course dates validation too strict

### ISSUE THREAD

**Issue by krusche** [ls1intum/Artemis#9410]
### Describe the bug

I would like to start a course before the enrollment is possible.

### To Reproduce

Set the start date before the enrollment date

### Expected behavior

In general, the validation should not be too strict

### Screenshots

<img width="1023" alt="image" src="https://github.com/user-attachments/assets/c599e9f8-b6d8-43ea-a546-17f09bf690be">

### Which version of Artemis are you seeing the problem on?

7.5.5

### What browsers are you seeing the problem on?

Chrome

**Comment by krusche** [comment]
Another issue, even if I disable student course enrollment, I cannot save anymore due to the too strict validation

<img width="1015" alt="image" src="https://github.com/user-attachments/assets/547e4dbf-5055-44b9-b4f6-3f80e002a30a">

**Comment by MaximilianAnzinger** [comment]
@krusche Why would you like a course to start before students can enroll?
Instructors, Tutors, and Editors have access to the resources prior to the actual teaching interval (Course start and end dates). This period should be used to signal students when the course takes place. Having one that takes place but isn't accessible seems like an anti pattern.
Feel free to correct me if i'm wrong here.

**Comment by krusche** [comment]
Example: I would like to see the course in the iOS app in order to communicate with the tutors beforehand

**Comment by krusche** [comment]
I don't think we need to validate the relation between enrollment and start date, let's give the instructor the freedom to configure this however he/she wants

**Comment by JohannesStoehr** [comment]
In this case we should rename "start date" to "visible date" or something along this line. Otherwise students will be confused why this particular course already starts earlier than other courses and how they can join it.

**Comment by MaximilianAnzinger** [comment]
And we should remove that date from the course information.

**Comment by ValentinHerrmann** [comment]
Start date before the start of enrollment is now possible. The core of the issue is resolved.
